src/App.tsx

```diff
@@ -1,8 +1,83 @@
-import React from 'react';
+import React, { useState, FC } from 'react';
+import 'bootswatch/dist/lux/bootstrap.min.css';
+import { getCorrectTodos } from './utils/api';
+import { TodoList } from './components/TodoList';
+import { Controller } from './components/Controller';
+
 import './App.css';
 
-const App = () => (
-  <h1>Dynamic list of TODOs</h1>
-);
+const App: FC = () => {
+  const [todoList, setTodoList] = useState<PreparedTodo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleClick = () => {
+    getCorrectTodos().then(list => {
+      setIsLoading(true);
+
+      // setTimeout for demo only
+      setTimeout(() => {
+        setTodoList(list);
+        setIsLoading(false);
+      }, 500);
+    });
+  };
+
+  const idSorter = () => {
+    setTodoList([...todoList]
+      .sort((a, b) => a.id - b.id));
+  };
+
+  const titleSorter = () => {
+    setTodoList([...todoList]
+      .sort((a, b) => a.title.localeCompare(b.title)));
+  };
+
+  const userNameSorter = () => {
+    setTodoList([...todoList]
+      .sort((a, b) => {
+        if (a.user && b.user) {
+          return a.user.name.localeCompare(b.user.name);
+        }
+
+        return 0;
+      }));
+  };
+
+  const completedSorter = () => {
+    setTodoList([...todoList]
+      .sort((a, b) => {
+        const isCompletedA = a.completed ? 1 : 0;
+        const isCompletedB = b.completed ? 1 : 0;
+
+        return isCompletedA - isCompletedB;
```

Comment:
их можно привести к число легче
```suggestion
        return +a.completed - b.completed;
```

Reply:
+a.completed - Number(b.completed); Мне удалось только так преобразовать, поскольку ТайпСкрипт против чтобы я производила вычисления числа с булем.. 
плюс перед b.completed в скобках не заработало
Number(a.completed) - Number(b.completed мне кажется лучше для восприятия

Reply:
я опечатался, очевидно что нужно было оба привести к число :)